### PR TITLE
Refs #8710 - created katello-debug sub-package

### PR DESCRIPTION
**Pull request by lzap**
With review of katello-debug plugin script, did a cleanup.

@jsherrill @ehelms can you guys give me assistance in changing composes,
building and pushing the subpackage change out?

I also want to make a change to the installer to install `katello-debug`
package on Capsules. What is the best way to do that @iNecas?

Thanks

### CODE REVIEW

**Comment by stbenjam** [comment]
For now, you'll want to add it to `puppet-capsule`... but in the future we need to have a katello-capsule package that requires this, had a restart script, and anything else capsule-specific.

File: deploy/script/katello-debug.sh [hunk 1]
```diff
@@ -5,8 +13,23 @@ then
   exit 1
 fi
 
-# General stuff
-add_files "/var/log/audit/audit.log"
+# Installer
+add_files "/var/log/katello-installer/*"
+add_files "/etc/katello-installer/*"
```
Comment (stbenjam) [inline deploy/script/katello-debug.sh]: Due to the installer break out, {/etc,/var/log}/capsule-installer and {/etc,/var/log}/sam-installer as well

**Comment by lzap** [comment]
Oh I figured out where the compose files are :-) Nice and handy.

http://koji.katello.org/koji/taskinfo?taskID=218610

**Comment by lzap** [comment]
I was thinking about puppet-capsule and started actually the file, but I figured out it is unnecessary. Foreman-debug helper functions ignores files/dirs which does not exist. Therefore you can run `katello-debug` both on Server and Capsule and both return reasonable tarballs. I can split the file into two, but I think that is little bit over-complicated.

File: deploy/script/katello-debug.sh (hunk 1 above)
Comment (lzap) [inline deploy/script/katello-debug.sh]: Oh good catch, amended.

**Comment by stbenjam** [comment]
Ah, I meant to use `puppet-capsule` to install the package on the capsule because there's no dedicated package yet, unless you want to add one here to the katello-spec.

You could also  add it to capsule-installer, although I'd prefer it to be separate.

**Comment by jlsherrill** [comment]
if this is just for the capsule, any reason to not call it 'capsule-debug'?

**Comment by stbenjam** [comment]
I can think of a few reasons why the prefix is better, it comes up in a search for `rpm -qa | grep katello`, it will automatically work with katello-remove, etc.

**Comment by stbenjam** [comment]
And I believe it's for both, one debug package for capsule and katello.

**Comment by lzap** [comment]
Yeah, it works for both. You run the same script on Server and Capusle and it gives reasonable results for both. Splitting is an overkill IMHO.

**Comment by jlsherrill** [comment]
Ah i see, l LGTM

**Comment by jlsherrill** [comment]
[test]

File: deploy/script/katello-debug.sh [hunk 2]
```diff
@@ -5,8 +13,23 @@ then
...
+add_files "/var/log/katello/*"
+add_files "/etc/pulp/server/plugins.d/*"
+add_files "/etc/foreman/plugins/katello.yaml"
+
+# Splice
+add_files "/var/log/splice/*"
+add_files "/etc/splice/*"
+add_files "/etc/httpd/conf.d/splice.conf",
+add_files "/etc/cron.d/spacewalk-sst-sync"
+add_files "/etc/cron.d/splice-sst-sync"
```
Comment (ehelms) [inline deploy/script/katello-debug.sh]: Do we still need these?

File: deploy/script/katello-debug.sh [hunk 3]
```diff
@@ -5,8 +13,23 @@ then
...
-# General stuff
-add_files "/var/log/audit/audit.log"
+# Installer
+add_files "/var/log/{katello,capsule,sam}-installer/*"
+add_files "/etc/{katello,capsule,sam}-installer/*"
+add_cmd "find /root/ssl-build -ls | sort -k 11" "katello_ssl_build_dir"
+add_cmd "find /etc/pki -ls | sort -k 11" "katello_pki_dir"
+
+# Katello
+add_files "/var/log/katello/*"
```
Comment (ehelms) [inline deploy/script/katello-debug.sh]: I don't think we have any of our own log files that exist here anymore.

File: deploy/script/katello-debug.sh (hunk 2 above)
Comment (jlsherrill) [inline deploy/script/katello-debug.sh]: they can be removed

Comment (lzap) [inline deploy/script/katello-debug.sh]: Both 31-32?

**Comment by lzap** [comment]
Amended the three lines.

**Comment by lzap** [comment]
Hello?

**Comment by lzap** [comment]
Please when merging this also merge this one: https://github.com/Katello/puppet-capsule/pull/33/files

Downstream BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1134109

Thanks

**Comment by jlsherrill** [comment]
[test]

**Comment by jlsherrill** [comment]
ACK from me, @stbenjam ?

**Comment by stbenjam** [comment]
Yea ACK

**Comment by jlsherrill** [comment]
Thanks @lzap

**Pull request merged**